### ISSUE TITLE
The library gives me only -90 to +90 roll when pitch has -180 to +180 range

### ISSUE THREAD

**Issue by Nikolay19971997** [xioTechnologies/Fusion#23]
Can you please tell me if there is anything wrong with my settings for this library. I rotate my bmi160 chip for roll and it stops on -90 and when keep rotating it in the same direction it is going back to zero with the same sign. The same goes for +90. It does not happen to pitch, it has range from 180 to -180 and working ok. Yaw has no problem with the range despite the fact that my board has no magnetometer. I had some problem calculating sensitivity in degrees per second per lsb and sensitivity in g per lsb due to no experience with mems sensors but i don't think it would influence the program this way. Even if i calculated it wrong all axis would behave simillar i think, not this way.

```
samplePeriod = 0.04f;

	gyroscopeSensitivity.axis.x = 0.0609756097560976f;
	gyroscopeSensitivity.axis.y = 0.0609756097560976f;
	gyroscopeSensitivity.axis.z = 0.0609756097560976f;

	accelerometerSensitivity.axis.x = 0.0001220703125f;
	accelerometerSensitivity.axis.y = 0.0001220703125f;
	accelerometerSensitivity.axis.z = 0.0001220703125f;

	FusionBiasInitialise(&fusionBias, 4.0f, samplePeriod);
	FusionAhrsInitialise(&fusionAhrs, 0.5f);


      void get_angles(float *Roll,float *Pitch,float *Yaw,struct bmi160_sensor_data *accel,struct bmi160_sensor_data *gyro)
{
	FusionVector3` uncalibratedGyroscope = {
	.axis.x =  gyro->x, /* replace this value with actual gyroscope x axis measurement in lsb */
	.axis.y =  gyro->y, /* replace this value with actual gyroscope y axis measurement in lsb */
	.axis.z =  gyro->z, /* replace this value with actual gyroscope z axis measurement in lsb */
	};

	FusionVector3 calibratedGyroscope = FusionCalibrationInertial(uncalibratedGyroscope, FUSION_ROTATION_MATRIX_IDENTITY, gyroscopeSensitivity, FUSION_VECTOR3_ZERO);


	// Calibrate accelerometer
	FusionVector3 uncalibratedAccelerometer = {
	   .axis.x = accel->x, /* replace this value with actual accelerometer x axis measurement in lsb */
	   .axis.y = accel->y, /* replace this value with actual accelerometer y axis measurement in lsb */
	   .axis.z = accel->z, /* replace this value with actual accelerometer z axis measurement in lsb */
	};
	FusionVector3 calibratedAccelerometer = FusionCalibrationInertial(uncalibratedAccelerometer, FUSION_ROTATION_MATRIX_IDENTITY, accelerometerSensitivity, FUSION_VECTOR3_ZERO);

	calibratedGyroscope = FusionBiasUpdate(&fusionBias, calibratedGyroscope);

	// Update AHRS algorithm
	FusionAhrsUpdateWithoutMagnetometer(&fusionAhrs, calibratedGyroscope, calibratedAccelerometer, samplePeriod);

	FusionEulerAngles eulerAngles = FusionQuaternionToEulerAngles(FusionAhrsGetQuaternion(&fusionAhrs));

	*Roll = eulerAngles.angle.roll;
	*Pitch = eulerAngles.angle.pitch;
	*Yaw = eulerAngles.angle.yaw;

}
```

**Comment by xioTechnologies** [comment]
You can verify your accelerometer sensitivity by checking that the output is 1 g when the sensor is stationary on a flat surface.  You can verify your gyroscope sensitivity by setting the algorithm gain to zero (so that the accelerometer is ignored) and checking that the measured changes in orientation match the actual changes in orientation.  For example, if you rotate the 90 degrees around one axis.

**Comment by Nikolay19971997** [comment]
Thank you for your reply. I did everything you recommended, but it didn't solve the problem. It seems i calculated the setting ok. I tried to set gain to zero the system behaved the same, i even turned of the gyroscope and it showed the same problem. I meant is it even ok that it shows only -90 to +90 range on y axes? because x and z shows -180 to 180 range. It meets my requirements for my task because i am not going to rotate it that much but it seems like there is a problem with calculations. But your library works perfect anyway
